cli/src/potpie/server_manager.py

```
@@ -0,0 +1,425 @@
+                    stderr=logout,
+                )
+            if celery_process.returncode not in [0, None]:
+                raise StartServerError(
+                    f" Start Celery Failed: with error: {celery_process.stderr}"
+                )
+
+            with open(self.pid_file, "a+") as f:
+                f.write(str(celery_process.pid))
+        except Exception as e:
+            logging.error(f"Failed to start Celery worker: {e}")
+            raise StartServerError(f"Failed to start Celery worker: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)